homeassistant/setup.py

```diff
@@ -211,8 +212,11 @@ def log_error(msg: str) -> None:
                       platform_name, msg)
         async_notify_setup_error(hass, platform_path)
 
-    integration = await loader.async_get_integration(hass, platform_name)
-
+    try:
+        integration = await loader.async_get_integration(hass, platform_name)
+    except IntegrationNotFound:
+        log_error("Integration not found")
+        return None
     if not integration:
```

Comment:
We can remove this check

Reply:
config entry will throw out in one of the test. It expects this function not raise error